src/app/auth/auth.service.ts

```diff
@@ -15,27 +21,51 @@ interface UserInfo {
 export class AuthService {
   public user$: Observable<firebase.User>;
 
-  constructor(private firebaseAuth: AngularFireAuth) {
+  constructor(
+    private firebaseAuth: AngularFireAuth,
+    public dialogRef: MatDialog,
+    private router: Router,
+  ) {
     this.user$ = firebaseAuth.authState;
   }
 
-  public login({ email, password }: UserInfo): Promise<firebase.auth.UserCredential> {
-    return this.firebaseAuth
-      .auth
-      .signInWithEmailAndPassword(email, password);
+  public isAuthenticated(): Observable<boolean> {
+    return this.user$.pipe(
+      map(user => !!user),
+    );
   }
 
-  public logout(): Promise<void> {
-    return this.firebaseAuth.auth.signOut();
+  public login(): void {
+    this.dialogRef.open(LoginComponent)
+    .afterClosed()
+    .pipe(
+      filter(user => !!user),
+      switchMap(({ email, password }: UserInfo) => {
```

Comment:
Maybe it would be better to get name of interface `UserCredentials` instead

Reply:
hmm.. yeah, you're right `UserInfo` isn't such a meaningful name. However, this interface includes 'displayed name' witch isn't part of Credentials.